### PR TITLE
1 construct the environment

### LINKED ISSUE #1: Construct the environment
Description:

Construct the environment required for the RL agent to interact with and be trained for a pole balancing task. 

Steps of implementation:

- [x] Set up the environment parameters (cart mass, pole mass, gravity, friction of the surface, pole length)
- [x] Define the state space. The space will include four variables: cart's position, cart velocity, pole angle and velocity
- [x] Create state transition function. This is to enable the update of the environment to update according to the agent's action
- [x] Define the reward function. This will return a numeric reward for the agent each time the agent takes an action. A positive reward will be given if the agent balances the pole and negative reward otherwise. 
- [ ] Set up a termination condition. This will be when the pole-cart setup travels balanced without requiring the agent to balance it any further
- [ ] Create episode method which incorporates the state-action-reward cycle defined above. The loop of episodes terminates either when the termination condition is satisfied or if the maximum number of steps is reached (the latter is an optional parameter)
- [ ] Set up a reset method (to reset the method and start from scratch again)

### PR DESCRIPTION
**Pull request by omar-emara**
@lucyfarnik can you add a list of change of conventions that you would like me to make for easier integration between the environment and the RL training algorithm? Also please review the branch and comment on any issues within it. I will update the branch according when you comment back on these points and then it will be ready to merge.

### CODE REVIEW

**Comment by omar-emara** [comment]
@lucyfarnik a step function has been added to better suit the training development it has been done. Code tested. Please review and merge if happy.

File: construct_pole_balancing_env.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+from gymnasium.envs.classic_control.cartpole import CartPoleEnv
+
+class ConstructPoleBalancingEnv(CartPoleEnv):
+    """
+    Construtor function to create the environment for the RL task for pole-cart
+    problem. 
+    
+    - Initialise object by calling ConstructPoleBalancingEnv()
+    while training:
```
Comment (lucyfarnik) [inline construct_pole_balancing_env.py:23]: Update this docstring

File: construct_pole_balancing_env.py [hunk 2]
```diff
@@ -0,0 +1,207 @@
+"""
+This module contains the functions required to construct the environment for 
+the RL agent to interact with. The following methods are included:
+    - env_params_setup (cart mass, pole mass, gravity, friction of the surface, pole length))
```
Comment (lucyfarnik) [inline construct_pole_balancing_env.py:4]: This looks out of date, am I missing something?

File: construct_pole_balancing_env.py [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+        
+        pole_velocity : float
+            defaults to stationary
+            
+        max_iter : int
+            default to 100 iterations
+            
+        iterations: int
+            starts as 0 and increments with each time state_transition is called
+
```
Comment (lucyfarnik) [inline construct_pole_balancing_env.py:90]: Add in new params

**Review by lucyfarnik — APPROVED** [review]
Some of the docstrings are out of date, other than that lgtm

**Pull request merged**